### PR TITLE
refactor ckpt and add topk save

### PR DESCRIPTION
**Pull request by samsja**
What this pr do:

* refactor ckpt args/config into its own nested BaseConfig
* allow to pass resume as a boolean that will pick the latest available ckpt
* allow only to keep topk ckpt to keep reduce space in  file system / s3 / gcp storage
* refactor and move ckpt related code to ckpt_utils.y

### CODE REVIEW

File: open_diloco/train_fsdp.py [hunk 1]
```diff
@@ -228,11 +214,11 @@ def train(config: Config):
         log_visible_maddrs(dht.get_visible_maddrs(), only_p2p=False)
 
     if local_rank == 0:
-        check_checkpoint_path_access(config.checkpoint_path, rank, config.hv.world_rank if config.hv else None)
+        check_checkpoint_path_access(config.ckpt.path, rank, config.hv.world_rank if config.hv else None)
 
     # DataLoader preparation
     tokenizer = AutoTokenizer.from_pretrained("mistralai/Mistral-7B-v0.1", use_fast=True)
-    tokenizer.pad_token = "</s>"  # Ensure pad token is set for models that need it
+    tokenizer.pad_token = " { "  # Ensure pad token is set for models that need it
```
Comment (Jackmin801) [inline open_diloco/train_fsdp.py:221]: is there a reason for changing the pad token? this seems like a weird pad token to me

**Review by Jackmin801 — CHANGES_REQUESTED** [review]
nice! makes the checkpointing stuff more modular and the new resume logic should make it that one doesnt need to change the launch string.

added a comment on a change that looks a bit weird to me

File: open_diloco/train_fsdp.py (hunk 1 above)
Comment (samsja) [reply open_diloco/train_fsdp.py:221]: wtf good catch. I think it is coming from cursor refactoring stuff haha

**Review by Jackmin801 — APPROVED** [review]
nice! lgtm

**Pull request merged**